smarsy/parse.py

```diff
@@ -121,3 +121,84 @@ def login():
                                     get_user_credentials(),
                                     get_headers())
     return response
+
+
+def bs_safeget(soup, *args):
+    """
+        Utility function used to get a content string from a
+        Beautiful Soup object and tuple of selectors. Returns False
+        if no object is found for the given selector
+        """
+    if len(args) == 1:
+        selectedElems = soup.select(args[0])
+    else:
+        for arg in args:
+            selectedElems = soup.select(arg)
+    if selectedElems is not None and len(selectedElems) > 0:
+        return selectedElems[0]
+    return False
+
+
+def get_parents_img(parent_html, *args):
+    """
+    Function receive: HTML, tags for search and return parents img URL
+    """
+    return bs_safeget(parent_html, *args)
+
+
+def get_parents_name(parent_html, *args):
+    """
+    Function receive: HTML, tags for search and return parents full name
+    """
+    return bs_safeget(parent_html, *args)
+
+
+def get_parents_b_date(parent_html, *args):
+    """
+    Function receive: HTML, tags for search and return parents b_date
+    """
+    return bs_safeget(parent_html, *args)
+
+
+def create_parents_dict(parent_html) -> dict:
+    """
+    Function receive HTML and return parent dictionary with given keys
+    """
+    parent_dict = {}
+    parent_list = []
+    parents_keys = ['parent_img', 'parent_name', 'parent_surname',
+                    'parent_middlename', 'parent_type',
+                    'parent_birth_date']
+    parent_img = get_parents_img(parent_html, '[valign=top]', 'img[src]')
+    if parent_img:
+        parent_list.append(parent_img.attrs['src'])
+    parents_fullname = get_parents_name(parent_html, '.username')
+    if parents_fullname:
+        parents_name = parents_fullname.text.split(' ')
+        parent_list.append(parents_name[0])
+        parent_list.append(parents_name[1])
+        parent_list.append(parents_name[2])
+        parent_list.append(parents_name[3][1:-1])
+    parent_b_date = get_parents_b_date(parent_html, '.userdata')
+    if parent_b_date:
+        parent_list.append(
+            str(convert_to_date_from_russian_written(parent_b_date.text)))
+    for parents_key, parent_value in zip(
+            parents_keys, parent_list):
+        parent_dict[parents_key] = parent_value
+    return parent_dict
+
+
+def parent_page_content_to_object(html) -> list:
+    try:
+        soup = BeautifulSoup(html, 'html.parser')
+        parent_tab = bs_safeget(soup, 'table')
+        parents = []
+        if parent_tab:
+            for parent in parent_tab.children:
+                parents.append(create_parents_dict(parent))
+        else:
```

Comment:
what's the point of this else block? If you think that there is a reason of having it then test is missing :)

Reply:
deleted else block so that in any case an empty list is returned